source/m3_bind.c

```diff
@@ -147,3 +147,72 @@ M3Result  m3_LinkRawFunction  (IM3Module            io_module,
 {
     return FindAndLinkFunction (io_module, i_moduleName, i_functionName, i_signature, (voidptr_t)i_function, LinkRawFunction);
 }
+
+// --------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+IM3Function  FindFunction    (IM3Module       io_module,
+                                    ccstr_t         i_moduleName,
+                                    ccstr_t         i_functionName,
+                                    ccstr_t         i_signature)
+{
+    M3Result result = m3Err_functionLookupFailed;
+
+    bool wildcardModule = (strcmp (i_moduleName, "*") == 0);
+    
+    for (u32 i = 0; i < io_module->numFunctions; ++i)
+    {
+        IM3Function f = & io_module->functions [i];
+
+        if (f->import.moduleUtf8 and f->import.fieldUtf8)
+        {
+            if (strcmp (f->import.fieldUtf8, i_functionName) == 0 and
+               (wildcardModule or strcmp (f->import.moduleUtf8, i_moduleName) == 0))
+            {
+                return f;
+            }
+        }
+    }
+
+    return NULL;
+}
+
+M3Result  LinkRawFunctionEx  (IM3Module io_module,  IM3Function io_function, ccstr_t signature,  const void * i_function, uint64_t cookie)
+{
+    M3Result result = m3Err_none;                                                 d_m3Assert (io_module->runtime);
+
+_try {
+_   (ValidateSignature (io_function, signature));
+
+    IM3CodePage page = AcquireCodePageWithCapacity (io_module->runtime, 2);
```

Comment:
should be 3

Reply:
oh sorry, was to quick reviewing my modifs, I pushed an updated version of the code. Thanks.